plugins/python-build/bin/python-build

```diff
@@ -480,7 +480,13 @@ reuse_existing_tarball() {
 }
 
 download_tarball() {
-  local package_url="$1"
+  #replaces python sources url
+  if [ -n "$PYENV_PYTHON_MIRROR_URL" ]; then
+      local package_url="$(echo "$1" | sed -r "s#(.*//www.python.org/ftp/python)*#$PYENV_PYTHON_MIRROR_URL#")"
```

Comment:
This only works for CPython versions.

pyenv can install numerous other types including Jython, PyPy, Stackless, miniconda, etc. and this solution will fail silently if you try to install one of those versions from a mirror

Perhaps a better solution would be one where the entire URL and base path of the python source/binary is replace